tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_OverlapHelpers.cpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+
+namespace LinearSolver::Schwarz {
+
+SPECTRE_TEST_CASE("Unit.ParallelSchwarz.OverlapHelpers",
+                  "[Unit][ParallelAlgorithms][LinearSolver]") {
+  {
+    INFO("Overlap extents");
+    CHECK(overlap_extent(3, 0) == 0);
+    CHECK(overlap_extent(3, 1) == 1);
+    CHECK(overlap_extent(3, 2) == 2);
+    CHECK(overlap_extent(3, 3) == 2);
+    CHECK(overlap_extent(3, 4) == 2);
+    CHECK(overlap_extent(0, 0) == 0);
+    CHECK(overlap_extents(Index<1>{{{3}}}, 2, 0) == Index<1>{{{2}}});
+    CHECK(overlap_extents(Index<2>{{{3, 4}}}, 2, 0) == Index<2>{{{2, 4}}});
+    CHECK(overlap_extents(Index<2>{{{3, 4}}}, 2, 1) == Index<2>{{{3, 2}}});
+    CHECK(overlap_extents(Index<3>{{{3, 4, 5}}}, 2, 0) ==
+          Index<3>{{{2, 4, 5}}});
+    CHECK(overlap_extents(Index<3>{{{3, 4, 5}}}, 2, 1) ==
+          Index<3>{{{3, 2, 5}}});
+    CHECK(overlap_extents(Index<3>{{{3, 4, 5}}}, 2, 2) ==
+          Index<3>{{{3, 4, 2}}});
```

Comment:
Test a case where the volume extent is the most restrictive limit.

Reply:
I actually realized the `overlap_extents` function is not necessary, so I removed the function in a fixup commit. All it did was replace the volume extent in one of the dimensions with the overlap extent; the interface I'm building will be cleaner without this function.